src/java/org/apache/cassandra/gms/EndpointState.java

```diff
@@ -138,6 +143,12 @@ void markDead()
         isAlive = false;
     }
 
+    public boolean isEmpty()
+    {
+        Map<ApplicationState, VersionedValue> state = applicationState.get();
+        return hbState.isEmpty() && !(state.containsKey(ApplicationState.STATUS_WITH_PORT) || state.containsKey(ApplicationState.STATUS));
```

Comment:
if we call `nodetool assassinate` we do not want to have host replacement be able to replace the host, so need this logic (no status found) to not pull this case into the unknown case.

* restart cluster but leave 1 node down
* assassinate downed node
* replace downed node

this will cause a gossip node to have version=-1 but will have the LEFT status; when we send this via gossip shadow round we don't want the receiver to think its a empty state, we want the recover to know about LEFT